code/API_definitions/number-recycling.yaml

```
@@ -0,0 +1,279 @@
+
+  version: wip
+  x-camara-commonalities: 0.4.0
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+externalDocs:
+  description: Product documentation at Camara
+  url: https://github.com/camaraproject/NumberRecycling
+servers:
+  - url: '{apiRoot}/number-recycling/vwip'
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root, defined by the service provider, e.g. `api.example.com` or `api.example.com/somepath`
+paths:
+  /check:
+    post:
+      security:
+        - openId:
+            - number-recycling:check
+      tags:
+        - Check Number Recycling
+      description: Check whether the subscriber of the phone number has changed.
+      operationId: checkNumberRecycling
+      parameters:
+        - $ref: '#/components/parameters/x-correlator'
+      requestBody:
+        description: |
+          Check whether the subscriber of the phone number has changed.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateCheckNumRecycling'
+        required: true
+      responses:
+        '200':
+          $ref: '#/components/responses/Generic200'
+        '400':
+          $ref: '#/components/responses/Generic400'
+        '401':
+          $ref: '#/components/responses/Generic401'
+        '403':
+          $ref: '#/components/responses/Generic403'
+        '404':
+          $ref: '#/components/responses/Generic404'
+        '422':
+          $ref: '#/components/responses/Generic422'
+        '500':
+          $ref: '#/components/responses/Generic500'
+        '503':
+          $ref: '#/components/responses/Generic503'
+components:
+  securitySchemes:
+    openId:
+      type: openIdConnect
+      openIdConnectUrl: https://example.com/.well-known/openid-configuration
+  parameters:
+    x-correlator:
+      name: x-correlator
+      in: header
+      description: Correlation id for the different services
+      schema:
+        type: string
+  headers:
+    x-correlator:
+      description: Correlation id for the different services
+      schema:
+        type: string
+  schemas:
+    CreateCheckNumRecycling:
+      type: object
+      required:
+        - specifiedDate
+      properties:
+        phoneNumber:
+          type: string
+          pattern: '^\+[1-9][0-9]{4,14}$'
+          example: '+346661113334'
+          description: A public identifier addressing a telephone subscription. In mobile networks it corresponds to the MSISDN (Mobile Station International Subscriber Directory Number). In order to be globally unique it has to be formatted in international format, according to E.164 standard, prefixed with '+'.
```

Reference separate schema
```suggestion
        phoneNumber:
          $ref: '#/components/schemas/PhoneNumber'
```

---

Thanks. Reference separate schema is added.